runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -3396,6 +3425,664 @@ J9::Z::TreeEvaluator::ArrayCopyBNDCHKEvaluator(TR::Node * node, TR::CodeGenerato
    return NULL;
    }
 
+/*
+ * \brief
+ *    Return the node representing the value to be written for an indirect wrtbar
+ *
+ * \note
+ *    For address type using compressedrefs, the compressed refs sequence is skipped
+ *    from the sub tree and return the uncompressed address value.
+ *    For all other cases, return the second child
+ *
+ */
+static TR::Node *getIndirectWrtbarValueNode(TR::Node *node, TR::Compilation *comp)
+   {
+   TR_ASSERT(node->getOpCode().isIndirect() && node->getOpCode().isWrtBar(), "getIndirectWrtbarValueNode expects indirect wrtbar nodes only n%dn (%p)\n", node->getGlobalIndex(), node);
```

Comment:
If this assertion were to fail, is the code below safe? If not let's make the assertion fatal.

Reply:
The asserts in this routine were made fatal here:
https://github.com/eclipse/openj9/pull/4743/commits/791f785f9aac893bec17b5d401842fc757b6d6f3